### PR TITLE
New module: Manage processes via pm2

### PR DESCRIPTION
**Pull request by 10sr**
##### SUMMARY

Manage processes via [pm2](http://pm2.keymetrics.io/docs/usage/quick-start/).

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

`lib/ansible/modules/web_infrastructure/pm2.py`

##### ANSIBLE VERSION

```
ansible 2.4.0 (pm2_module e5da0b5863) last updated 2017/08/06 15:59:37 (GMT +900)
  config file = /Users/yuk/my/j/ansible/ansible.cfg
  configured module search path = [u'/Users/yuk/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/yuk/my/ansible/lib/ansible
  executable location = /Users/yuk/my/ansible/bin/ansible
  python version = 2.7.12 (default, Oct 22 2016, 01:11:08) [GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
```

### CODE REVIEW

**Comment by 10sr** [comment]
ready_for_review

File: lib/ansible/modules/web_infrastructure/pm2.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {'status': ['preview'],
```
Comment (jtyr) [inline lib/ansible/modules/web_infrastructure/pm2.py]: I would format this like this:

```
ANSIBLE_METADATA = {
    'status': ['preview'],
    'supported_by': 'community',
    'metadata_version': '1.0'
}
```

File: lib/ansible/modules/web_infrastructure/pm2.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: pm2
+author:
+    - "10sr (@10sr)"
```
Comment (jtyr) [inline lib/ansible/modules/web_infrastructure/pm2.py]: I suggest you use your real name here.

File: lib/ansible/modules/web_infrastructure/pm2.py [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+version_added: "2.4"
+short_description: Manage processes via pm2
+description:
+    - Manage the state of processes via pm2 process manager.
```
Comment (jtyr) [inline lib/ansible/modules/web_infrastructure/pm2.py]: In `EXAMPLES` you use indentation of 2 characters. I suggest you do the same in the `DOCUMENTATION` and `RETURN` as well.

File: lib/ansible/modules/web_infrastructure/pm2.py [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+        default: null
+        description:
+            - Change into this directory before running pm2 start command.
+            - When I(state=started) and this option is omitted, use the
+              directory where I(config) or I(script) exists.
+'''
+
+EXAMPLES = '''
+---
+# Start myapp with process config file, if not running
```
Comment (jtyr) [inline lib/ansible/modules/web_infrastructure/pm2.py]: I would use these comments as a name of the task:

```
- name: Start myapp with process config file, if not running
  pm2:
    name: myapp
    config: /path/to/myapp/myapp.json
    state: started
```

**Review by jtyr — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/web_infrastructure/pm2.py (hunk 2 above)
Comment (10sr) [reply lib/ansible/modules/web_infrastructure/pm2.py]: I don't like to add my real name here, so I want to keep it as it is if possible...

**Comment by 10sr** [comment]
ready_for_review

**Comment by 10sr** [comment]
ready_for_review

**Comment by 10sr** [comment]
bot_status

**Comment by jtyr** [comment]
I did not test it but from the point of view of the code/documentation review I'm giving it `shipit`.

**Comment by 10sr** [comment]
bot_status

**Comment by 10sr** [comment]
@jtyr Sorry to bother you but it seems you are "requesting changes". Could you resolve this so that ansibot would remove `needs_revision` label?

**Comment by jtyr** [comment]
I don't see how to do that. You have addressed all my comments which cleared all my comments from the code.

**Comment by 10sr** [comment]
I'm not sure precisely, too...
How about doing `approve` from here?

image

**Review by jtyr — APPROVED** [review]

**Comment by 10sr** [comment]
Thanks!!!!!

**Pull request closed without merging**